src/prefect/deployments.py

```diff
@@ -499,25 +499,24 @@ def handle_openapi_schema(cls, value):
     @classmethod
     @sync_compatible
     async def load_from_yaml(cls, path: str):
-        with open(str(path), "r") as f:
-            data = yaml.safe_load(f)
-
-            # load blocks from server to ensure secret values are properly hydrated
-            if data["storage"]:
-                block_doc_name = data["storage"].get("_block_document_name")
-                # if no doc name, this block is not stored on the server
-                if block_doc_name:
-                    block_slug = data["storage"]["_block_type_slug"]
-                    block = await Block.load(f"{block_slug}/{block_doc_name}")
-                    data["storage"] = block
-
-            if data["infrastructure"]:
-                block_doc_name = data["infrastructure"].get("_block_document_name")
-                # if no doc name, this block is not stored on the server
-                if block_doc_name:
-                    block_slug = data["infrastructure"]["_block_type_slug"]
-                    block = await Block.load(f"{block_slug}/{block_doc_name}")
-                    data["infrastructure"] = block
+        data = yaml.safe_load(await anyio.Path(path).read_bytes())
```

Comment:
Not only should we not be performing blocking IO here, we should also not keep the file open while loading blocks from the API.